lib/sort_by_length.rb

```diff
@@ -1,7 +1,30 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: This is an insertion sort with two loops, so time complexity is O(n^2) / quadratic
+# Space complexity: O(1) / constant
```

Comment:
Technically here since you're doing `.split` you're making a new array of words.  So this is O(n)